rootconf/default/etc/init.d/discover.sh

```diff
@@ -68,6 +69,7 @@ do_start() {
             install_remain_sticky_arch || {
                 echo "Error: problems making install boot mode sticky" > /dev/console
             }
+            sync;sync
```

Comment:
I don't think so.  u-boot env variables are kept in a mtd device, not a mtdblock device.  sync applies to *filesystems*.  There is no filesystem on a mtd device.